networkit/csbridge.py

```diff
@@ -21,132 +22,9 @@ def widget_from_graph(G, node_scores = None, node_partition = None, node_palette
-
-	# Set color palettes (maybe overwritten below)
-	if node_palette is not None:
-		palette = node_palette
-	else:
-		if not have_seaborn:
-			raise MissingDependencyError("seaborn")
-		palette = seaborn.color_palette("rocket_r", as_cmap=True).colors
-
-	# Color calculation: score = continuous distribution, partition = discrete distribution
-	hc_colors = []
-
-	# Partition
-	if node_partition is not None:
-		if node_palette is None:
-			palette = seaborn.color_palette("hls", node_partition.numberOfSubsets())
-		else:
-			if len(node_palette) < node_partition.numberOfSubsets():
-				raise Exception("Number of partitions higher than number of colors in provided palette. Provide node_palette with enough colors.")
-
-		partitions = node_partition.getVector()
-
-		if len(palette) < node_partition.numberOfSubsets():
-			raise Exception("Number of partitions to high for default coloring. Provide node_palette with enough colors.")
-
-		for i in G.iterNodes():
-			hc_colors.append((palette[partitions[i]][0] * 255, palette[partitions[i]][1] * 255, palette[partitions[i]][2] * 255))
-
-	# Score
-	elif node_scores is not None:
-
-		minhc = min(node_scores)
-		maxhc = max(node_scores)
-
-		# calculate coloring of nodes
-		def get_rgb(minimum, maximum, value):
-			minimum, maximum, value = float(minimum), float(maximum), float(value)
-			ratio = int((len(palette) - 1) * (value-minimum) / (maximum - minimum) * (value-minimum) / (maximum - minimum))
-			r = int(palette[ratio][0] * 255)
-			g = int(palette[ratio][1] * 255)
-			b = int(palette[ratio][2] * 255)
-			return r, g, b
-
-		if abs(maxhc - minhc) > 0:
-			for score in node_scores:
-				hc_colors.append(get_rgb(minhc, maxhc, score))
-		else:
-			color = palette[int((len(palette) -1) / 2)];
-			for i in G.iterNodes():
-				hc_colors.append((color[0] * 255, color[1] * 255, color[2] * 255))
-
-	# No node values
-	else:
-		color = palette[0];
-		for i in G.iterNodes():
-			hc_colors.append((color[0] * 255, color[1] * 255, color[2] * 255))
-	
-	# Set styling
-	if show_ids:
-		s = [{
-			'selector': 'node',
-				'css': {
-					'background-color': 'data(color)',
-					'content': 'data(id)'
-				}}]
-	else:
-		s = [{
-			'selector': 'node',
-				'css': {
-					'background-color': 'data(color)'
-				}}]
-
-	# Create widget
-	cytoWidget = ipycytoscape.CytoscapeWidget()
-
-	nodes = []
-	edges = []
-
-	if G.isDirected():
-		edge_class = "directed "
-	else:
-		edge_class = "undirected "
-
-	for i in G.iterNodes():
-		n = ipycytoscape.Node(data={"id": i, "color": hc_colors[i] })
-		nodes.append(n)
-
-	for u,v in G.iterEdges():
-		e = ipycytoscape.Edge(data={"source": u, "target": v, "classes": edge_class })
-		edges.append(e)
-
-	# It is much faster to add edges and nodes in bulk.
-	cytoWidget.graph.add_nodes(nodes)
-	cytoWidget.graph.add_edges(edges, G.isDirected())
 
-	cytoWidget.set_style(s)
-	cytoWidget.set_layout(name='cose')
-	return cytoWidget
```

Comment:
I recommend adding a deprecation warning here, users will probably not read the code/docs.